parse/parser_internal_test.go

```diff
@@ -1,7 +1,67 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package parse
 
 import "testing"
 
+type MockTokenBuffer struct {
+	buf []Token
+	sp  int
+}
+
+func (m *MockTokenBuffer) Read() Token {
+	ret := m.buf[m.sp]
+	m.sp++
+	return ret
+}
+
+func (m *MockTokenBuffer) Peek(n peekNumber) Token {
+	return m.buf[m.sp+int(n)]
+}
+
+func TestParse_curTokenIs_nextTokenIs(t *testing.T) {
+	tokenBuf := MockTokenBuffer{
```

Comment:
What about separating curTokenIs, nextTokenIs test case?

Reply:
First time I separated these two test case functions, but in that case there are duplicated code like making tokenBuf code. So I combined curTokenIs and nextTokenIs test case. But, how about other people's think?

Reply:
Then, how about making a function like `setupMockTokenBuffer() TokenBuffer`, `buildMockTokenBuffer() TokenBuffer` in test file. And just call and sign it to `tokenBuf`!

```go
func TestParse_curTokenIs(t *testing.T) {
    tokenBuf := setupMockTokenBuffer()
    ret := curTokenIs(&tokenBuf, Int)
    ...
}
```

Like this!

Reply:
Right. I will do it.